src/engraving/libmscore/score.cpp

```diff
@@ -2795,10 +2795,40 @@ KeyList Score::keyList() const
     Staff* firstStaff = nullptr;
     for (Staff* s : masterScore()->staves()) {
         if (!s->isDrumStaff(Fraction(0, 1))) {
+            static bool firstStaff = true;
             KeyList* km = s->keyList();
-            tmpKeymap.insert(km->begin(), km->end());
-            firstStaff = s;
-            break;
+            if (firstStaff) {
+                firstStaff = false;
+                tmpKeymap = *km;
+                firstStaff = s;
```

Comment:
Personally I'm confused by this section... two variables with the same name? A static variable?

Reply:
Sorry, pushed a bad version, i have pushed the fix for that and also fixed what i think xcode was angry about